### PR TITLE
Bug fixes, add atmos watertank

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Fixes watertanks (https://github.com/ParadiseSS13/Paradise/issues/421) and updates extinguisher code.
2) Adds the atmos watertank, which is able to fire nanofrost, metal foam and water (https://github.com/tgstation/-tg-station/pull/5645).
3) Cleans up some of the AI/robot code, may fix https://github.com/ParadiseSS13/Paradise/issues/419.
4) Fixes IPC heads sticking through hardsuits.
5) Fixes trays (https://github.com/ParadiseSS13/Paradise/issues/412). Trays are now bags rather than their own custom thing.
6) Fixes the position of the (follow) link in Robot Talk.
7) Plant-B-Gone now affects Diona Nymphs (or any monkey with "plant" as their mutantrace).
8) Dionaea now heal fire damage, to fix https://github.com/ParadiseSS13/Paradise/issues/321. They only heal fire damage at 1/4th of the speed at which they heal oxygen damage (so generally less than 1 per tick), and now only heal brute damage at 1/2 the speed of oxygen damage to compensate.
9) Dionaea should no longer always have the overfed icon on their HUD.
10) Fixes https://github.com/ParadiseSS13/Paradise/issues/424.

### CODE REVIEW

**Comment by Markolie** [comment]
@Fox-McCloud Could you have a look at the water reagent/nanofrost change? I tested it and it works well, but seeing as you're more familiar with ZAS/LINDA differences than I you may have some feedback.

**Comment by fj45** [comment]
Firefighting gear sounds awesome, hopefully it works alright.

**Comment by Markolie** [comment]
It'll work about as well as an extinguisher does right now, due to ZAS.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -1713,7 +1722,7 @@ datum
 
 			nanites
 				name = "Nanites"
-				id = "nanities"
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Reagents.dm]: Ahahahah.

**Comment by Fox-McCloud** [comment]
Nanofrost is pretty darn cool, mechanics wise, but I have a feeling it won't work well on our server because of ZAS.

Basically NanoFrost puts out the fires it touches instantly, coverts the areas hit into nitrogen, and chills the area---this is amazing, in effect, with tile-based atmos system, but for ZAS, it's not going to work with large zones and turn small zones into frozen death areas.

In any event, I still support this being added; we're going with LINDA, eventually, so it only makes sense; we just might have to remove access to the atmos backpack at first (we'll have to see how it goes) if it turns out to be a murder machine for small zones.

**Comment by fj45** [comment]
Shame, but if the nano-frost is a murdermachine, maybe we could have it as a (short lived) atmos traitor item? Could be rather !FUN!

**Comment by Fox-McCloud** [comment]
@fj45 

While that sounds like a good idea, I'd rather not cement it in players' heads that it's a way to murder people as opposed to using it to put out fires.

**Comment by Spacemanspark** [comment]
Eh, it would only be for a little bit, if you're really going to port LINDA.

**Comment by Markolie** [comment]
It's worth noting that nanofrost will never cool anything below 0 degrees, so it'll always be safe for humans.

**Comment by Deanthelis** [comment]
Can the backpack be filled with blood, space lube, etc. like a normal extinguisher can (or could)?

**Comment by Markolie** [comment]
Nope.

**Pull request merged**